src/base/rss/utorrentrssdata.cpp

```diff
@@ -0,0 +1,215 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "utorrentrssdata.h"
+
+#include <libtorrent/lazy_entry.hpp>
+
+#include <QDebug>
+#include <QFile>
+#include <QStringList>
+
+#ifdef QBT_USES_QT5
+#include <QRegularExpression>
+#else
+#include <QRegExp>
+#endif
+
+#include "rssfeed.h"
+#include "rssdownloadrule.h"
+#include "rssdownloadrulelist.h"
+#include "base/utils/string.h"
+
+using namespace Rss;
+using namespace libtorrent;
+using namespace Utils::String;
+
+namespace
+{
+    int POSTPONE_OPTS[] = { 0, 0, 0, 1, 2, 3, 4, 7, 14, 21, 30 };
+
+    enum EnabledFlags
+    {
+        ENABLED      = 1,
+        RAW_NAME     = 2,
+        MAX_PRIORITY = 4,
+        SMART_FILTER = 8,
+        PAUSED       = 16
+    };
+
+    enum Quality
+    {
+        QUALITY_ALL   = -1,
+        QUALITY_480p  = 0x10000,
+        QUALITY_720p  = 0x00800,
+        QUALITY_1080i = 0x01000,
+        QUALITY_1080p = 0x02000
+    };
+
+    void applyFilter(Rss::DownloadRule *rule, const libtorrent::lazy_entry &filter)
+    {
+#ifdef QBT_USES_QT5
+        // Leading/Trailing *'s and *'s before/after '|'
+        const QRegularExpression STRIP_PATTERN(R"((?<=^|\|)\*+|\*+(?=$|\|))");
+#else
+        // Leading/Trailing *'s and *'s before '|'
+        // No lookbehind, so as close as it'll get without complicating things
+        const QRegExp STRIP_PATTERN(R"(^\*+|\*+(?=$|\|))");
+#endif
+
+        rule->setUseRegex(true);
+        rule->setMustNotContain(fromStdString(filter.dict_find_string_value("not_filter")).remove(STRIP_PATTERN).replace('*', ".*").replace('?', "."));
+
+        QString mustContain = fromStdString(filter.dict_find_string_value("filter")).remove(STRIP_PATTERN).replace('*', ".*").replace('?', ".");
+
+        QStringList qualityList;
+        int quality = filter.dict_find_int_value("quality");
+        if (quality != QUALITY_ALL) {
+            if (quality & QUALITY_480p)  qualityList << "480p";
+            if (quality & QUALITY_720p)  qualityList << "720p";
+            if (quality & QUALITY_1080i) qualityList << "1080i";
+            if (quality & QUALITY_1080p) qualityList << "1080p";
```

Comment:
This project is only for importing settings from uTorrent, it has nothing to do with the type of content that is actually available. uTorrent only has those 5 options.